ts/features/bonus/cgn/screens/CgnDetailScreen.tsx

```diff
@@ -320,3 +244,100 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
 });
 
 export default connect(mapStateToProps, mapDispatchToProps)(CgnDetailScreen);
+
+export function CgnCardStatus({ card }: { card: Card }) {
+  return (
+    <React.Fragment>
+      {CardRevoked.is(card) && (
+        <Tag
+          testID="card-status-revoked"
+          variant="error"
+          text={I18n.t("bonus.cgn.detail.status.badge.revoked")}
+        />
+      )}
+      {CardExpired.is(card) && (
+        <Tag
+          testID="card-status-expired"
+          variant="error"
+          text={I18n.t("bonus.cgn.detail.status.badge.expired")}
+        />
+      )}
+      {CardActivated.is(card) && (
+        <Chip color="grey-650" testID="card-bottom-content">
+          {I18n.t("bonus.cgn.detail.status.date.valid_until", {
+            date: formatDateAsShortFormat(card.expiration_date)
+          })}
+        </Chip>
+      )}
+    </React.Fragment>
+  );
+}
```

Comment:
What do you think about extracting this component in another file just to make this file more readable and splitting the context? Even if this component is only used here, it could be useful in other iterations and also it can be tested better.

Reply:
I would not move it to another file. The only reason it is a component and not inlined is to not modify the existing tests too much since I do not have the knowledge of requirements to do that. Also creating many small files adds to cognitive load, and adds unnecessary double checking when changing code when checking for dead code. Also I would prefer to not write code for the future. I took me longer to code this whole change request because of unnecessarily parametrized code that obviously could not account for the future requirements.

Reply:
I understand your points, but I believe that [modularization](https://en.wikipedia.org/wiki/Modular_programming) is a best practice that we should adhere to. By keeping this component in a separate file, we enhance the modularity of the code, which has several benefits. The modularization improves code maintainability, readability, and reusability.

Keeping components in separate files also simplifies testing. When a component is isolated, it can be tested independently, leading to more focused and efficient tests. This separation also makes it easier to identify and fix issues without affecting unrelated parts of the code. Additionally, modular code is easier to extend and refactor as the project evolves, which helps prevent the accumulation of technical debt.

While it might initially seem like an added cognitive load, the long-term benefits of modularization, such as easier debugging, better testability, and clearer organization, outweigh the short-term convenience of having everything in a single file. Ultimately, modular code supports better software development practices and results in a more scalable and maintainable codebase.

Reply:
done